yaml_overview.rst

```diff
@@ -2,17 +2,60 @@ YAML Overview
 +++++++++++++
 
 Before writing CloudSlang code it helps to have a good working knowledge
-of YAML. This section contains a brief overview of some of the YAML
-specification. See the full `YAML
-specification <http://www.yaml.org/spec/1.2/spec.html>`__ for more
-information.
+of YAML. YAML is a human friendly, data serialization language that has
+become a popular choice for configuration files and other hand-crafted
+data files. CloudSlang uses YAML to define its flows and operations.
 
-Whitespace
-==========
+This section contains a brief overview of the common YAML syntax and the
+best practices for writing it. See the full `YAML specification
+<http://www.yaml.org/spec/1.2/spec.html>`__ for more information.
+
+Basics
+======
+
+The contents of a YAML file define a single data structure (graph) that
+is composed of nested nodes (mappings, sequences and scalars). Well
+crafted YAML files are easy to read, and can have comments in the right
+places to explain things better.
+
+YAML tries to be human friendly, by minimizing the need for special
+syntax characters when the data is simple. The most common YAML special
+characters in YAML are:
+
+- ``:`` - Between key/value pairs
+- ``-`` - Denotes a sequence entry
+- ``#`` - Starts a comment
+
+NOTE: You should be aware that YAML also supports very complex data, and
```

Comment:
Please use Sphinx note directive: http://www.sphinx-doc.org/en/stable/markup/para.html#directive-note here and in several places below.